### PR TITLE
Turn on noUnusedLocals

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
A few notes:
1. `verifyTelemetryProperties` wasn't used in the configure tests, but seems somewhat useful so I added it and fixed the failing tests
1. the cpp generation was ignoring `exposeStatements`, but I assume that's fine since it says `// We don't open a port for Cpp` a few lines above. Updated the cpp test to reflect this

### CODE REVIEW

File: commands/azureCommands/pull-from-azure.ts [hunk 1]
```diff
@@ -91,18 +87,3 @@ async function pullImage(loginServer: string, imageRequest: string, username: st
 
     terminal.sendText(`docker pull ${loginServer}/${imageRequest}`);
 }
-
-async function isLoggedIntoDocker(loginServer: string): Promise<{ configPath: string, loggedIn: boolean }> {
```
Comment (PrashanthCorp) [inline commands/azureCommands/pull-from-azure.ts]: Is this an unused function?

File: debugging/coreclr/dockerDebugConfigurationProvider.ts [hunk 2]
```diff
@@ -100,7 +100,7 @@ export class DockerDebugConfigurationProvider implements DebugConfigurationProvi
 
         const { appFolder, resolvedAppFolder } = await this.inferAppFolder(folder, debugConfiguration);
 
-        const { appProject, resolvedAppProject } = await this.inferAppProject(folder, debugConfiguration, resolvedAppFolder);
+        const { resolvedAppProject } = await this.inferAppProject(folder, debugConfiguration, resolvedAppFolder);
```
Comment (PrashanthCorp) [inline debugging/coreclr/dockerDebugConfigurationProvider.ts]: I don't see changes to the `inferAppProject` function to reflect this change.

File: commands/azureCommands/pull-from-azure.ts (hunk 1 above)
Comment (EricJizbaMSFT) [reply commands/azureCommands/pull-from-azure.ts]: Yes

File: debugging/coreclr/dockerDebugConfigurationProvider.ts (hunk 2 above)
Comment (EricJizbaMSFT) [reply debugging/coreclr/dockerDebugConfigurationProvider.ts]: `inferAppProject` returns an object - removing `appProject` is essentially like removing a property name from an interface, which doesn't require any sort of change on the original object.

**Comment by EricJizbaMSFT** [comment]
Related issue: https://github.com/microsoft/vscode-docker/issues/848

**Comment by EricJizbaMSFT** [comment]
ping @nturinski @PrashanthCorp @StephenWeatherford

File: debugging/coreclr/dockerManager.ts [hunk 3]
```diff
@@ -60,11 +60,6 @@ type LastImageBuildMetadata = {
     options: DockerBuildImageOptions;
 };
 
-type LastContainerRunMetadata = {
```
Comment (StephenWeatherford) [inline debugging/coreclr/dockerManager.ts]: fyi @philliphoff

File: debugging/coreclr/netCoreProjectProvider.ts [hunk 4]
```diff
@@ -2,13 +2,12 @@
...
  *--------------------------------------------------------*/
 
-import * as path from 'path';
 import { DotNetClient } from "./dotNetClient";
 import { FileSystemProvider } from "./fsProvider";
 import { TempFileProvider } from './tempFileProvider';
 
 const getTargetPathProjectFileContent =
-`<Project>
+    `<Project>
```
Comment (StephenWeatherford) [inline debugging/coreclr/netCoreProjectProvider.ts]: Why this change?  Personally I prefer them all to be left-indented if any part of the multi-line string is.

File: test/configure.test.ts [hunk 5]
```diff
@@ -143,6 +143,8 @@ async function testConfigureDocker(platform: Platform, expectedTelemetryProperti
             }
         }
     }
+
+    verifyTelemetryProperties(actionContext, expectedTelemetryProperties);
```
Comment (StephenWeatherford) [inline test/configure.test.ts]: Thx.

File: debugging/coreclr/netCoreProjectProvider.ts (hunk 4 above)
Comment (EricJizbaMSFT) [reply debugging/coreclr/netCoreProjectProvider.ts]: My only change was to remove the `path` import above. This must have been auto-formatted

Comment (EricJizbaMSFT) [reply debugging/coreclr/netCoreProjectProvider.ts]: Yeah this is getting changed on-save. I like it better left-indented, too, but don't feel like fighting the auto format
